### ISSUE TITLE
Investigate cmake component based installer to generate seperate package for clients

### ISSUE THREAD

**Issue by preda** [ROCmSoftwarePlatform/rocFFT#116]
Please provide a compiled application that exercises the library (similar to clFFT-client), in the release package (rocfft.deb) or as a separate package.

There should be an easy way to test and benchmark the library.

**Comment by tingxingdong** [comment]
rocfft provides a benchmarking client called rider 
https://github.com/ROCmSoftwarePlatform/rocFFT/wiki/Clients

**Comment by preda** [comment]
Yes, but it's not included in the release .deb package, or in a pre-built package of its own. The user needs to build from sources to get "rider".

**Comment by tingxingdong** [comment]
we won't release the benchmarking client in .deb, as it is not library, but is only a utility tool.

**Comment by preda** [comment]
OK, fine, please provide some precompiled version of the client, separate from the library. I think clFFT has a separate package for the client. The idea is to be able to do some sequence of "adb install"s and be able to run something afterwards.

**Comment by kknox** [comment]
@preda 
Yes, in the old opencl days, we used to package a client, but over time we realized that most users don't want the client, and it litters the system with utilities that aren't used.  We did end up removing the client from the opencl packages too.  

I do understand your request, it would be nice for our test teams to be able to install the test clients from a separate package.  The only thing stopping this featuers, is that I don't have the knowledge yet to get cmake to create multiple packages on a 'make package'.  It tends to stick everything in one .deb.  However, I've been investigating/reading about the cmake component installs, and think that may be the way forward, but just haven't had the time yet.  Lets keep this issue open to track the desire for a component based installer.
